tools/fetch-sponsors.js

```diff
@@ -387,6 +391,57 @@ async function fetchGitHubSponsors() {
 	};
 }
 
+/**
+ * Fetches thanks.dev data using the REST API.
+ * @returns {Promise<{sponsors: Array}>} An promise that resolves to an object with one property:
+ *  - `sponsors` (Array): List of sponsors
+ */
+async function fetchThanksDevData() {
+	const endpoint = "https://api.thanks.dev/v1/entity/gh/eslint/donors";
+
+	const { body: result } = await request(endpoint, {
+		method: "GET",
+		headers: { "Content-Type": "application/json" },
+	});
+
+	const payload = await result.json();
+
+	if (process.env.DEBUG) {
+		await fs.writeFile(
+			"./debug-thanksdev-raw-response.json",
+			JSON.stringify(payload, null, 4),
+			{ encoding: "utf8" },
+		);
+	}
+
+	const sponsors = Object.values(payload.donors).reduce(
```

Comment:
I think this may be clearer by doing a `filter()` and then a `map()`. First filter out all of the donors that don't have a donation in the current month, then map into the final form.

`reduce()` is really for calculating a single value based on an array of values.

Reply:
updated